### DISCUSSION TITLE
Correct cell not found

### DISCUSSION THREAD

**Discussion by norafried** [Parcels-code/Parcels#1264] [Q&A]
I'm using POP and want to release particles at different depths for about two years:

When I'm executing my code I get the following message:

Correct cell not found for (-7.563470, 63.168785) after 1000000 iterations
Debug info: old particle indices: (yi, xi) 207 399
            new particle indices: (yi, xi) 207 0
            Mesh 2d shape:  401 401
            Relative particle position:  (xsi, eta) 1.00264628104553

What exactly does it mean and can I do something to stop it? It only happens after a certain amount of time.

**Comment by erikvansebille** [comment]
Hi @norafried. This error occurs when Parcels can't find in which cell a particle is located. This sometimes happens when grids are highly curved. In your case, it seems to happen on the zonal boundary (zonal cell was `399` first, then `0`). So my hunch would be that this happens when your particles cross your zonal reentrant boundary.

You could have a look at this discussion (for the Hycom grid): https://github.com/OceanParcels/parcels/issues/1240. A solution that worked in that case is at https://github.com/OceanParcels/parcels/issues/1240#issuecomment-1275098832. Perhaps it works in your case too?

**Reply by norafried** [reply]
Hi @erikvansebille! Thanks for your answer. I tried implementing the sane two lines but still get the same error. I commented on the recent discussion and maybe that will help.

**Reply by erikvansebille** [reply]
Ah, if these two extra lines don't work, then it's likely another problem. The error you got was different from the one originally reported in #1240. 

But can you confirm that this _is_ a problem on the domain boundary, as I interpreted from the error?

**Reply by norafried** [reply]
How can I figure out it's on the domain boundary? For sure my data is limited to the subpolar North Atlantic. So in my head it makes sense that there are particles trying to leave the area. Would make sense to just end their path? I don;t really care where they leave to I would say?

**Reply by erikvansebille** [reply]
You can plot the trajectories?

**Reply by norafried** [reply]
Yes I can. I get a normal output and 'normal' trajectories. Therefore I was wondering whether it is an error or more a warning.

**Reply by erikvansebille** [reply]
OK, I didn't realise this error only happens very incidentally; I thought that you got a lot of these and that it was hampering your analysis...

As I said above, the error occurs when Parcels can't find in which cell a particle is located. This sometimes happens when grids are highly curved (e.g. close to a pole). See https://github.com/OceanParcels/parcels/blob/master/parcels/include/index_search.h#L318-L452 for the full code, and then in particular the code below for the error message https://github.com/OceanParcels/parcels/blob/56e988ae0f627ec8994c656f5f4bab71154a2e47/parcels/include/index_search.h#L406-L412
You could try to increase the `maxIterSearch` parameter, but that typically doesn't help in my experience. 

The problem likely is that the grid cells are so skewed and the particle is so close to an edge, that within numerical precision, it's correct grid cell can't be found. So there's not much you can do about it, the particle will be deleted. If it occurs only a handful of times, I would not worry about it and simply ignore the particles (they will be shorter in the output file)

**Reply by norafried** [reply]
Okay, thanks. I will see how many time it occurs with a longer run. It did occur more than a handful of times. But I guess depending on how many particles you release that number might scale up.
